src/lib/snack-bar/snack-bar-ref.ts

```diff
@@ -84,13 +88,18 @@ export class MatSnackBarRef<T> {
       this._onAction.complete();
     }
 
-    this._afterClosed.next();
-    this._afterClosed.complete();
+    this._afterDismissed.next(this._dismissedWithAction);
+    this._afterDismissed.complete();
+    this._dismissedWithAction = false;
   }
 
-  /** Gets an observable that is notified when the snack bar is finished closing. */
-  afterDismissed(): Observable<void> {
-    return this._afterClosed.asObservable();
+  /**
+   * Gets an observable that is notified when the snack bar is finished closing.
+   * The stream emits a boolean that indicates whether the snack bar was dismissed
+   * using the action button.
+   */
+  afterDismissed(): Observable<boolean> {
```

Comment:
Instead of making this a boolean stream, I'd rather add an event object so that we could add more information to it in the future.